engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -153,12 +153,18 @@ public void deleteProcess(final ProcessRecord processRecord) {
 
     processColumnFamily.deleteExisting(processDefinitionKey);
     processByIdAndVersionColumnFamily.deleteExisting(idAndVersionKey);
-    digestByIdColumnFamily.deleteExisting(fkProcessId);
 
     processesByProcessIdAndVersion.remove(processRecord.getBpmnProcessIdBuffer());
     processesByKey.remove(processRecord.getProcessDefinitionKey());
-    versionManager.deleteProcessVersion(
-        processRecord.getBpmnProcessId(), processRecord.getVersion());
+
+    final long latestVersion =
+        versionManager.getCurrentProcessVersion(processRecord.getBpmnProcessId());
+
+    if (latestVersion == processRecord.getVersion()) {
+      digestByIdColumnFamily.deleteIfExists(fkProcessId);
```

Comment:
It shouldn't because we delete the digest instead of updating it. Imagine this:

1. Deploy 3 versions of the process
2. Delete version 3. The digest will be removed from the CF
3. Delete version 2. Version 2 is the latest now, but there is no digest for it.

Reply:
👍 Good point!

🔧 Please add a comment to this line explaining why we can't use `deleteExisting` 🙇